src/test/ui/coercion/coerce-just-metadata.rs

```diff
@@ -0,0 +1,23 @@
+// run-pass
```

Comment:
this doesn't need to be `run-pass`, probably just `build-pass` or `check-pass` if you don't care about exercising the codegen parts.

Reply:
also, could you add some negative tests in another file? I want to make sure that for some `Struct: !Trait`, we can't do a `JustMetadata<Struct>` coercion into `JustMetadata<dyn Trait>`, for example. Or like a unsize coercion of various weird combinations.

Reply:
This was run-pass while I was testing because it's easier to run just a single ui test than a single doctest 😅

It does need to be build-pass to actually codegen the unsizing, which is where the complexity is.